components/dashboard/src/service/public-api.ts

```diff
@@ -179,3 +184,32 @@ function createServiceClient<T extends ServiceType>(
         },
     });
 }
+
+export function watchPrebuild(
+    request: PartialMessage<WatchPrebuildRequest>,
+    cb: (prebuild: Prebuild) => void,
+): () => void {
+    const MAX_BACKOFF = 60000;
+    const BASE_BACKOFF = 3000;
+    let backoff = BASE_BACKOFF;
+    const abort = new AbortController();
+    (async () => {
+        while (!abort.signal) {
+            try {
+                for await (const response of prebuildClient.watchPrebuild(request, {
+                    signal: abort.signal,
+                })) {
+                    cb(response.prebuild!);
```

Comment:
reset backoff

Reply:
we should only reset if it was successful?

Reply:
i.e. probabl in for await then?